### PR TITLE
fix: #2519 - adds USDC price for currencies in liquidity pool creator

### PR DESCRIPTION
**Pull request by prakashsanker**
1. Not sure if I should be using `currencyA` or `currencyB`
2. Now sure what the use case is - could someone elaborate?
3. Screenshot
Screen Shot 2021-10-13 at 4 04 46 PM

### CODE REVIEW

**Comment by prakashsanker** [comment]
Fixes #2519

**Comment by prakashsanker** [comment]
@justindomingue  bumping this!

File: src/components/InputStepCounter/InputStepCounter.tsx [hunk 1]
```diff
@@ -140,6 +144,11 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
+  const baseCurrencyPriceInUSDC = useUSDCPrice(currencyB ?? undefined)
```
Comment (justindomingue) [inline src/components/InputStepCounter/InputStepCounter.tsx:147]: sorry to be pedantic but this is a bit hard to follow. Could you move it to a custom hook  at the top of this file? Something like

```
function useBaseCurrencyAmount(baseCurrency: Currency | undefined, quoteAmount: number) {
  const currencyUSDCPrice = useUSDCPrice(baseCurrency ?? undefined)
  if (!currencyUSDCPrice) return undefined
  return Number(currencyUSDCPrice.toSignificant(6)) * quoteAmount
}
```

File: src/pages/AddLiquidity/index.tsx [hunk 2]
```diff
@@ -137,7 +137,10 @@ export default function AddLiquidity({
     baseCurrency ?? undefined,
     existingPosition
   )
-
+  const usdcBaseCurrencyPriceInNumber = Number(baseCurrencyPriceInUSDC?.toSignificant(4))
```
Comment (justindomingue) [inline src/pages/AddLiquidity/index.tsx:140]: this could also be in its own hook for readability

**Review by justindomingue — COMMENTED** [review]
@willhennessy have we settled on an approach?

File: src/constants/tokens.ts [hunk 3]
```diff
@@ -165,6 +165,8 @@ export const UNI: { [chainId: number]: Token } = {
   [SupportedChainId.KOVAN]: new Token(SupportedChainId.KOVAN, UNI_ADDRESS[42], 18, 'UNI', 'Uniswap'),
 }
 
+export const STABLECOINS = [DAI.symbol, USDC.symbol, USDT.symbol]
```
[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/constants/tokens.ts:168]: oooh TIL about symbols not being unique. ty!!

File: src/utils/index.ts [hunk 4]
```diff
@@ -55,3 +56,10 @@ export function isTokenOnList(tokenAddressMap: TokenAddressMap, token?: Token):
 export function formattedFeeAmount(feeAmount: FeeAmount): number {
   return feeAmount / 10000
 }
+
+export function isEthStablecoinPair(currencyA: Currency | undefined, currencyB: Currency | undefined) {
+  const isEthAndStable = currencyA?.isNative && STABLECOINS.includes(currencyB?.symbol)
```
[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/utils/index.ts:61]: ooh if we should return for stable-stable pairs it might be easier to just to something like `isEthOrStable(currencyA) && isEthOrStable(currencyB)`

File: src/components/PositionPreview/index.tsx [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
-import { Break } from 'components/earn/styled'
+import { Trans } from '@lingui/macro'
 import { Currency } from '@uniswap/sdk-core'
-import CurrencyLogo from 'components/CurrencyLogo'
-import DoubleCurrencyLogo from 'components/DoubleLogo'
-import JSBI from 'jsbi'
-import { LightCard } from 'components/Card'
 import { Position } from '@uniswap/v3-sdk'
 import RangeBadge from 'components/Badge/RangeBadge'
+import { LightCard } from 'components/Card'
```
[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/components/PositionPreview/index.tsx:5]: yeah i think the sort order has since changed 🤔 looks diff on main

File: src/components/InputStepCounter/InputStepCounter.tsx [hunk 6]
```diff
@@ -144,10 +144,8 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
-  const baseCurrencyPriceInUSDC = useUSDCPrice(currencyB ?? undefined)
-  const baseCurrencyPriceInUSDCNumber = Number(baseCurrencyPriceInUSDC?.toFixed(4))
-  const baseCurrencyPriceIsNotNaN = !isNaN(baseCurrencyPriceInUSDCNumber)
   const quoteCurrencyAmount = Number(value)
+  const baseCurrencyUSDPrice = useCurrencyUSDPrice(currencyB ?? undefined, quoteCurrencyAmount)
```
[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/components/InputStepCounter/InputStepCounter.tsx:149]: ooh hmm a few typing issues are making this difficult to use the `CurrencyAmount` format: `currencyB` might be undefined, and `quoteCurrencyAmount` is a decimal instead of a `BigInt` as the constructor expects. Maybe cleaner to do it without `CurrencyAmount`?

[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/components/InputStepCounter/InputStepCounter.tsx:149]: oooh nvm, i found a helper function that gives me what i need: `tryParsedAmount` which converts the input string into `CurrencyAmount`!

File: src/pages/AddLiquidity/index.tsx [hunk 7]
```diff
@@ -137,11 +138,12 @@ export default function AddLiquidity({
     baseCurrency ?? undefined,
     existingPosition
   )
-  const usdcBaseCurrencyPriceInNumber = Number(baseCurrencyPriceInUSDC?.toSignificant(4))
-  const quoteCurrencyPrice =
-    usdcBaseCurrencyPriceInNumber * Number(invertPrice ? price?.invert().toSignificant(4) : price?.toSignificant(4))
-  const roundedQuoteCurrencyPrice = isNaN(usdcBaseCurrencyPriceInNumber) ? undefined : quoteCurrencyPrice.toFixed(2)
-  const showBaseCurrencyPrice = isEthStablecoinPair(baseCurrency ?? undefined, quoteCurrency ?? undefined)
+
+  const quoteCurrencyAmount = invertPrice ? price?.invert().toSignificant(6) : price?.toSignificant(6)
```
Comment (tinaszheng) [inline src/pages/AddLiquidity/index.tsx:142]: would love another set of eyes on this math here 👀  i believe this is accomplishing the same thing as before, but simpler?

File: src/components/InputStepCounter/InputStepCounter.tsx [hunk 8]
```diff
@@ -140,6 +146,10 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
+  const baseCurrencyUSDPrice = useCurrencyUSDPrice(tryParseAmount(value, currencyB))
+  const showBaseCurrencyUSDPrice =
+    isEthOrStablecoin(currencyA ?? undefined) && isEthOrStablecoin(currencyB ?? undefined)
```
[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/components/InputStepCounter/InputStepCounter.tsx:151]: ah yeah i think `currencyA` could be `null` so it's to cast that into `undefined` in those cases

[the remark this answers is not in the document]
Comment (tinaszheng) [reply src/components/InputStepCounter/InputStepCounter.tsx:151]: ah good catch!

File: src/components/InputStepCounter/InputStepCounter.tsx [hunk 9]
```diff
@@ -140,6 +146,9 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
+  const baseCurrencyUSDPrice = useCurrencyUSDPrice(tryParseAmount(value, currencyB))
+  const showBaseCurrencyUSDPrice = isEthOrStablecoin(currencyA) && isEthOrStablecoin(currencyB)
```
Comment (justindomingue) [inline src/components/InputStepCounter/InputStepCounter.tsx:150]: i would probably skip running `useCurrencyUSDPrice` if `showBaseCurrencyUSDPrice` is false by passing `undefined` to the hook in that case?

Comment (justindomingue) [reply src/components/InputStepCounter/InputStepCounter.tsx:150]: then L190 only needs to check `baseCurrencyUSDPrice` is defined

File: src/hooks/useCurrencyUSDPrice.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
```
Comment (justindomingue) [inline src/hooks/useCurrencyUSDPrice.ts:1]: isn't this `useUSDValue`? 

https://github.com/Uniswap/interface/blob/main/src/hooks/useUSDCPrice.ts#L58

File: src/pages/AddLiquidity/index.tsx [hunk 11]
```diff
@@ -137,6 +139,12 @@ export default function AddLiquidity({
     existingPosition
   )
 
+  const quoteCurrencyAmount = invertPrice ? price?.invert().toSignificant(6) : price?.toSignificant(6)
+  const quoteCurrencyPrice = useCurrencyUSDPrice(tryParseAmount(quoteCurrencyAmount, quoteCurrency ?? undefined))
+
+  const showQuoteCurrencyPrice =
```
Comment (justindomingue) [inline src/pages/AddLiquidity/index.tsx:145]: ditto. could probably skip `useCurrencyUSDPrice` if this is false

File: src/hooks/useCurrencyUSDPrice.ts (hunk 10 above)
Comment (tinaszheng) [reply src/hooks/useCurrencyUSDPrice.ts:1]: my god yes it is

Comment (tinaszheng) [reply src/hooks/useCurrencyUSDPrice.ts:1]: thank u 🤯

**Comment by BigsteppA1** [comment]
Ok

**Comment by tinaszheng** [comment]
after playing around with the change, it feels like showing the USDC price isn't as helpful since only showing for stablecoin pairs means that it will roughly always match the stablecoin amount anyway. going to close this issue for now - sorry @prakashsanker and thank you for your contribution!
